### PR TITLE
[p-queue] Upgrade to 3.0

### PR DESCRIPTION
**Pull request by mad-mike**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
only notable change in v 3.0: https://github.com/sindresorhus/p-queue/pull/35

- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/p-queue/index.d.ts [hunk 1]
```diff
@@ -93,6 +93,9 @@ declare namespace PQueue {
         concurrency?: number;
         autoStart?: boolean;
         queueClass?: QueueClassConstructor<O>;
+        intervalCap?: number;
+        interval?: number;
+        carryoverConcurrencyCount?: number;
```
Comment (evanshortiss) [inline types/p-queue/index.d.ts]:
```suggestion
        carryoverConcurrencyCount?: boolean;
```

**Review by evanshortiss — CHANGES_REQUESTED** [review]
Looks like `carryoverConcurrencyCount` should be a boolean based on the README for 3.0

**Comment by mad-mike** [comment]
@evanshortiss you're obviously right. Should be fixed now.

**Review by evanshortiss — APPROVED** [review]

File: types/p-queue/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for p-queue 2.3
+// Type definitions for p-queue 3.0
```
Comment (rbuckton) [inline types/p-queue/index.d.ts]: When upgrading the major version, please follow the steps outlined here: https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Pull request closed without merging**